poc/map/Jack.tbasic

```diff
@@ -119,35 +141,135 @@ Console_ShowWindow(%Console_SW_SHOWMAXIMIZED)
+          End If
+          
+          me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+          
+          'Random depth profile
+          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+        Next
+        
     	Case %DISTRIBUTED
     	
+    	distributedBase = (me.worldList(i).minimumTotalRiverWidth /(( me.worldList(i).ambiance.riverArmsCount + 1 ) / 2)) / me.worldList(i).ambiance.riverArmsCount
+    	distributedTotal = 0
+
+      for j as integer = 1 to me.worldList(i).ambiance.riverArmsCount
+        'Width
+        if j = me.worldList(i).ambiance.riverArmsCount Then
+          me.worldList(i).arms(j).width += iseven(assign(me.worldList(i).minimumTotalRiverWidth - distributedTotal, me.worldList(i).arms(j).width))
+        Else
+          me.worldList(i).arms(j).width += iseven(assign(Round(distributedBase * j,0),me.worldList(i).arms(j).width))
+          distributedTotal += me.worldList(i).arms(j).width
+        endif
+        
+        me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+        
+        'Random depth profile
+        me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+      Next
+        
     	Case %MOSTLYNARROW
     	
+        narrowCount = 0
+        narrowCurrent = 3
+        narrowTotal = 0
+        
+        for j as integer = 1 to me.worldList(i).ambiance.riverArmsCount
+          incr narrowCount
+          if narrowCount > 3 then
+          	narrowCount = 0
+          	narrowCurrent += 2
+          End If
+          'Width
+          if j = me.worldList(i).ambiance.riverArmsCount Then
+            me.worldList(i).arms(j).width += iseven(assign(me.worldList(i).minimumTotalRiverWidth - narrowTotal, me.worldList(i).arms(j).width))
+          else
+            me.worldList(i).arms(j).width = narrowCurrent
+            narrowTotal += narrowCurrent
+          endif
+          
+          me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+          
+          'Random depth profile
+          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+        Next
+        
     	Case %MOSTLYWIDE
     	
+        wideTotal = 0
+        
+        for j as integer = 1 to me.worldList(i).ambiance.riverArmsCount
+          'Width
+          if j = me.worldList(i).ambiance.riverArmsCount Then
+            me.worldList(i).arms(j).width += iseven(assign(me.worldList(i).minimumTotalRiverWidth - wideTotal, me.worldList(i).arms(j).width))
+          Else
+            me.worldList(i).arms(j).width += iseven(assign(ceil(me.worldList(i).minimumTotalRiverWidth / (me.worldList(i).ambiance.riverArmsCount + j - 1)), me.worldList(i).arms(j).width))
```

Comment:
Can you spot the error here ? ThinBasic throw an "Expected a variable but found something else."  (Token found : ME).

Reply:
I think there's a problem with _Assign()_ , I prepared [a minimal example in thinBasic's forum](https://www.thinbasic.com/community/showthread.php?13075-ASSIGN-error-in-UDT).

Reply:
Thank you for the example - it is a bug and it should work out of the box. We will have a look with Eros.

The workaround is to use DIM .. At syntax, to overlay virtual variable over ME.member memory and work with that.